crates/task-impls/src/consensus.rs

```diff
@@ -51,11 +51,13 @@ use crate::{
     },
 };
 /// Alias for the block payload commitment and the associated metadata.
-pub struct CommitmentAndMetadata<PAYLOAD: BlockPayload> {
+pub struct CommitmentAndMetadata<TYPES: NodeType> {
```

Comment:
Might be worth also updating this doc comment to explain the purpose of the fee data.